apps/ledger-live-mobile/src/components/RequiresBLE/AndroidRequiresLocationPermissions.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { ReactNode, useCallback, useEffect, useState } from "react";
+import LocationRequired from "../../screens/LocationRequired";
+import {
+  checkLocationPermission,
+  locationPermission,
+  requestLocationPermission,
+  RequestResult,
+} from "./androidBlePermissions";
+
+/**
+ * Renders an error if location is required & not available,
+ * otherwise renders children
+ */
+const AndroidRequiresLocationPermissions: React.FC<{
+  children?: ReactNode | undefined;
+}> = ({ children }) => {
+  const [requestResult, setRequestResult] = useState<RequestResult | null>(
+    null,
+  );
+  const [checkResult, setCheckResult] = useState<boolean | null>(null);
+
+  const { granted } = requestResult || {};
+
+  const requestPermission = useCallback(async () => {
+    let dead = false;
+    requestLocationPermission().then(res => {
+      if (dead) return;
+      setRequestResult(res);
+    });
+    return () => {
+      dead = true;
+    };
+  }, []);
+
+  const checkPermission = useCallback(async () => {
+    let dead = false;
+    checkLocationPermission().then(res => {
+      if (dead) return;
+      setCheckResult(res);
+    });
+    return () => {
+      dead = true;
+    };
+  }, []);
+
+  useEffect(() => {
+    requestPermission();
+  }, [requestPermission]);
+
+  if (!locationPermission) return <>{children}</>;
+
+  if (checkResult || granted) return <>{children}</>;
+  if (requestResult === null) return null; // suspense PLZ
+  return (
+    <LocationRequired errorType="unauthorized" onRetry={checkPermission} />
```

Comment:
the component renders a screen - don't know if it's possible and easy to move `LocationRequired` inside `components` and not in `screens` ? Are they any other screens that navigate to it ?

Reply:
it's rendered by the "RenderError" component depending on the error :pain-ultra-hd:

Reply:
arf

Reply:
Move it to `components` 👍

Reply:
thanks !